core/pages/CmsPage.js

```diff
@@ -0,0 +1,55 @@
+import EventBus from '@vue-storefront/core/compatibility/plugins/event-bus'
+import { htmlDecode } from '@vue-storefront/core/filters/html-decode'
+import Composite from '@vue-storefront/core/mixins/composite'
+
+export default {
+  name: 'CmsPage',
+  mixins: [Composite],
+  computed: {
+    pageTitle () {
+      return this.$store.state.cms_page.current ? this.$store.state.cms_page.current.title : ''
+    }
+  },
+  watch: {
+    '$route': 'validateRoute'
+  },
+  asyncData ({ store, route, context }) { // this is for SSR purposes to prefetch data
+    return new Promise((resolve, reject) => {
+      if (context) context.output.cacheTags.add(`cmsPage`)
+      store.dispatch('cms_page/single', {
+        value: route.params.slug
+      }).then(res => {
+        if (res) {
+          store.state.cms_page.current = res
+          EventBus.$emitFilter('cms_page-after-load', { store: store, route: route }).then(results => {
```

Comment:
we are slowly depreciating event bus, please don't base your solutions on it

Reply:
Removed